### PR TITLE
chore(docs): Update the "Gatsby on Linux" page

### PR DESCRIPTION
**Pull request by 1qk1**
## Changes
- Removed "TODO" stub from /docs/gatsby-on-linux/ page since it has content
- Make the links at the bottom visible
- Add 2 more learning resources (1 at the links and 1 about WSL)

## Related Issues

#15529

### CODE REVIEW

File: docs/docs/gatsby-on-linux.md [hunk 1]
```diff
@@ -64,10 +62,11 @@ Or to install all at the same time and approve `(y)` all installs:
 sudo apt update && sudo apt -y upgrade && sudo apt install build-essential && sudo apt install git && sudo apt install libpng-dev
```
Comment (pranshuchittora) [inline docs/docs/gatsby-on-linux.md]: Nice work, can you write all commands separately with the proper description, so that users know about what the command will do on execution.

Comment (pranshuchittora) [inline docs/docs/gatsby-on-linux.md]: These commands are specific to Debian distros, it would be great if you provide similar docs for other distros like cent, arch, etc.

File: docs/docs/gatsby-on-linux.md [hunk 2]
```diff
@@ -64,10 +62,11 @@ Or to install all at the same time and approve `(y)` all installs:
...
-[n]: https://github.com/tj/n
-[n-install]: https://github.com/mklement0/n-install
-[nvm]: https://github.com/creationix/nvm
-[bash startup]: https://github.com/Microsoft/WSL/issues/776#issuecomment-266112578
+- [Super detailed guide to making VSCode work with ESL from VSCode's docs website](https://code.visualstudio.com/docs/remote/wsl)
+- [windows store](https://www.microsoft.com/en-us/store/p/ubuntu/9nblggh4msv6)
+- [n](https://github.com/tj/n)
+- [n-install](https://github.com/mklement0/n-install)
+- [nvm](https://github.com/creationix/nvm)
+- [bash startup](https://github.com/Microsoft/WSL/issues/776#issuecomment-266112578)
```
Comment (pranshuchittora) [inline docs/docs/gatsby-on-linux.md]: One line description would be helpful before visiting. As the reader can pick what is useful or not for his/her context.

File: docs/docs/gatsby-on-linux.md [hunk 3]
```diff
@@ -4,12 +4,10 @@ title: Gatsby on Linux
 
 # Linux
 
-> This is a TODO. Help our community expand it.
-
-> Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your pull request gets accepted.
-
```
Comment (pranshuchittora) [inline docs/docs/gatsby-on-linux.md]: Pls, undo these changes, as WSL !== Linux. The idea is the same but the underlying things may differ a bit.

File: docs/docs/gatsby-on-linux.md (hunk 2 above)
Comment (1qk1) [reply docs/docs/gatsby-on-linux.md]: Hmmm since they are all mentioned in the article, I could explain them there. I just enabled the links!

File: docs/docs/gatsby-on-linux.md (hunk 1 above)
Comment (1qk1) [reply docs/docs/gatsby-on-linux.md]: Sure, and another thing I was wondering is, I have installed gatsby a couple times on both windows and linux, and never had to install build-essential (on linux) or windows-build-tools (on windows), but in the docs it says you have to install them, are they really necessary?

Comment (1qk1) [reply docs/docs/gatsby-on-linux.md]: I have never used them but I will look them up.

File: docs/docs/gatsby-on-linux.md (hunk 3 above)
Comment (1qk1) [reply docs/docs/gatsby-on-linux.md]: I know it's different, it was just mentioned in #15529 for the "TODO" part to be removed, I will add it again. I have only made some small changes in this file.

File: docs/docs/gatsby-on-linux.md [hunk 4]
```diff
@@ -64,10 +62,11 @@ Or to install all at the same time and approve `(y)` all installs:
...
 
-<!-- links -->
+### Additional links and resources
 
-[windows store]: https://www.microsoft.com/en-us/store/p/ubuntu/9nblggh4msv6
-[n]: https://github.com/tj/n
-[n-install]: https://github.com/mklement0/n-install
-[nvm]: https://github.com/creationix/nvm
-[bash startup]: https://github.com/Microsoft/WSL/issues/776#issuecomment-266112578
+- [Super detailed guide to making VSCode work with ESL from VSCode's docs website](https://code.visualstudio.com/docs/remote/wsl)
```
Comment (wardpeet) [inline docs/docs/gatsby-on-linux.md]: Small typo

```suggestion
- [Super detailed guide to making VSCode work with WSL from VSCode's docs website](https://code.visualstudio.com/docs/remote/wsl)
```

File: docs/docs/gatsby-on-linux.md (hunk 1 above)
Comment (wardpeet) [reply docs/docs/gatsby-on-linux.md]: I think you do need them for gatsby-plugin-sharp and probably some other plugins. I'm unsure if gatsby core needs it.

File: docs/docs/gatsby-on-linux.md (hunk 4 above)
Comment (1qk1) [reply docs/docs/gatsby-on-linux.md]: Yes! I will add more detailed instructions while adding instructions to the part 0 page! I could split them by package manager too, like mentioned there.

File: docs/docs/gatsby-on-linux.md (hunk 1 above)
Comment (sever1an) [reply docs/docs/gatsby-on-linux.md]: `sharp` ships with binaries for windows, no build tools required.

Comment (sever1an) [reply docs/docs/gatsby-on-linux.md]: http://sharp.pixelplumbing.com/en/stable/install/

File: docs/docs/gatsby-on-linux.md [hunk 5]
```diff
@@ -4,12 +4,10 @@ title: Gatsby on Linux
 
 # Linux
 
-> This is a TODO. Help our community expand it.
-
-> Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your pull request gets accepted.
-
 ## Windows Subsystem Linux (WSL)
 
+This guide assumes that you already have WSL installed with an working Linux distro. If you don't, follow [this guide from microsoft's site](https://docs.microsoft.com/en-us/windows/wsl/install-win10) to install WSL and a Linux distro of your choice.
```
Comment (sever1an) [inline docs/docs/gatsby-on-linux.md]: > with an working Linux
Should be 
with a working Linux

File: docs/docs/gatsby-on-linux.md (hunk 1 above)
Comment (sever1an) [reply docs/docs/gatsby-on-linux.md]: Sharp tries to use supplied binaries first but will sometimes fall back to compiling source. http://sharp.pixelplumbing.com/en/stable/install/

**Comment by laurieontech** [comment]
Hey @1qk1, wanted to check to make sure you had everything you needed to continue work on this. Thanks for your work thus far!

**Comment by laurieontech** [comment]
Hey @1qk1, let me know if you'd like someone else to take this over. I know you're working on another PR on this topic as well, so just wanted to touch base.

File: docs/docs/gatsby-on-linux.md (hunk 5 above)
Comment (laurieontech) [inline docs/docs/gatsby-on-linux.md]:
```suggestion
This guide assumes that you already have WSL installed with a working Linux distro. If you don't, follow [this guide from microsoft's site](https://docs.microsoft.com/en-us/windows/wsl/install-win10) to install WSL and a Linux distro of your choice.
```

**Comment by laurieontech** [comment]
Thanks for the changes! Since we haven't heard anything from @1qk1  I'm going to go ahead and continue working on this.

**Comment by laurieontech** [comment]
Would love comments from @jamo and @nicholascapo if you two have any suggestions/feedback.

File: docs/docs/gatsby-on-linux.md [hunk 6]
```diff
@@ -2,12 +2,128 @@
...
+### Ubuntu, Debian and other `apt` based distros
+
+Begin by updating and upgrading.
+
+```shell
+sudo apt update
+sudo apt -y upgrade
+```
+
+> Only use `-y` if you want to upgrade to the latest version of the software.
```
Comment (jamo) [inline docs/docs/gatsby-on-linux.md]: The `-y` iirc will just automatically move forward w/o asking user for confirmation.

File: docs/docs/gatsby-on-linux.md [hunk 7]
```diff
@@ -2,12 +2,128 @@
...
+
+Once `curl` is installed, you can use it to install `nvm`, which will manage `node` and all its associated versions.
+
+```shell
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.1/install.sh | bash
+```
+
+> Note that this is the current stable release of nvm. Full installation instructions and troubleshooting can be found at the [nvm GitHub page](https://github.com/nvm-sh/nvm)
+
+When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses `10.17.0`, but other version numbers can be used instead.
```
Comment (jamo) [inline docs/docs/gatsby-on-linux.md]: IIRC with `nvm` you can just ask it to install the major version, and it will automatically pick up the latest release, i.e. `nvm install 10`

Comment (jamo) [reply docs/docs/gatsby-on-linux.md]: This way we don't have to update the materials when a new minor/patch comes out

File: docs/docs/gatsby-on-linux.md [hunk 8]
```diff
@@ -2,12 +2,128 @@
...
+```shell
+sudo dnf install git
+```
+
+### Archlinux and other `pacman` based distros
+
+Begin by updating.
+
+```shell
+sudo pacman -Sy
```
Comment (jamo) [inline docs/docs/gatsby-on-linux.md]: @nicholascapo do you recall if we need this as a separate step for arch

File: docs/docs/gatsby-on-linux.md (hunk 6 above)
Comment (laurieontech) [reply docs/docs/gatsby-on-linux.md]: I stole that from below, but good point on fixing the wording and being clearer.

File: docs/docs/gatsby-on-linux.md (hunk 8 above)
Comment (laurieontech) [reply docs/docs/gatsby-on-linux.md]: It was needed when I ran it in docker fwiw

Comment (jamo) [reply docs/docs/gatsby-on-linux.md]: 👍

**Comment by 1qk1** [comment]
> Thanks for the changes! Since we haven't heard anything from @1qk1 I'm going to go ahead and continue working on this.

Yes sorry I haven't been in front of a pc in the last week or so. 
Also because it's the linux instructions I wanted to finish with all the suggestions from the part 0 page first, and add instuctions here in a similar fashion, sorry for not replying to the previous comments.
Did you add all the suggestions @laurieontech? Is anything more I can add/improve?

**Comment by laurieontech** [comment]
No worries! I did add the suggestions and made some additions to the Linux section. It's not verbatim what's in part 0 since I did some docker image testing. So if everyone agrees to this perhaps we can port some of this over to the part 0 PR as well.

**Comment by 1qk1** [comment]
> So if everyone agrees to this perhaps we can port some of this over to the part 0 PR as well.

Yes! While reading the changes I was thinking of porting the nvm instructions there, and add the official node instructions as a note if someone doesn't want to use nvm.

File: docs/docs/gatsby-on-linux.md (hunk 1 above)
Comment (1qk1) [reply docs/docs/gatsby-on-linux.md]: Marking as resolved since @laurieontech fixed this.

Comment (1qk1) [reply docs/docs/gatsby-on-linux.md]: Marking as resolved since @laurieontech fixed this.

**Comment by laurieontech** [comment]
@1qk1 Seems like a great solution to me. We're likely going to edit the formatting on those instructions, so let's wait to get those 100% and merged and then we can port them over here?

**Comment by 1qk1** [comment]
> @1qk1 Seems like a great solution to me. We're likely going to edit the formatting on those instructions, so let's wait to get those 100% and merged and then we can port them over?

Sure, sounds good!

**Comment by 1qk1** [comment]
> @1qk1 Seems like a great solution to me. We're likely going to edit the formatting on those instructions, so let's wait to get those 100% and merged and then we can port them over?

Sure, sounds good!

**Comment by 1qk1** [comment]
So basically this is ready to be merged? It says "changes requested" but those changes were fixed by @laurieontech.

**Comment by laurieontech** [comment]
Not quite yet! I'm waiting for a second set of eyes on my changes. But it's on the list.

File: docs/docs/gatsby-on-linux.md [hunk 9]
```diff
@@ -2,12 +2,126 @@
 title: Gatsby on Linux
 ---
 
-> This is a TODO. Help our community expand it.
+# Linux
```
Comment (marcysutton) [inline docs/docs/gatsby-on-linux.md]: We can omit this, as it's duplicating the page header. That way, there will be only one h1 per page (the frontmatter title generates one already).
```suggestion
```

File: docs/docs/gatsby-on-linux.md [hunk 10]
```diff
@@ -2,12 +2,126 @@
...
 
-> Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your pull request gets accepted.
+This guide assumes you already have a native installation of Linux on your machine. The following steps walk through how to install `node` and associated depdendencies.
```
Comment (marcysutton) [inline docs/docs/gatsby-on-linux.md]:
```suggestion
This guide assumes you already have a native installation of Linux on your machine. The following steps walk through how to install Node.js and associated dependencies.
```

File: docs/docs/gatsby-on-linux.md [hunk 11]
```diff
@@ -2,12 +2,126 @@
...
+
+### Ubuntu, Debian and other `apt` based distros
```
Comment (marcysutton) [inline docs/docs/gatsby-on-linux.md]: This should be an h2, since it follows the h1.
```suggestion
## Ubuntu, Debian, and other `apt` based distros
```

File: docs/docs/gatsby-on-linux.md [hunk 12]
```diff
@@ -2,12 +2,126 @@
...
+
+Once `curl` is installed, you can use it to install `nvm`, which will manage `node` and all its associated versions.
+
+```shell
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.1/install.sh | bash
+```
+
+> Note that this is the current stable release of nvm. Full installation instructions and troubleshooting can be found at the [nvm GitHub page](https://github.com/nvm-sh/nvm)
+
+When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses the latest release of version `10`, but other version numbers can be used instead.
```
Comment (marcysutton) [inline docs/docs/gatsby-on-linux.md]:
```suggestion
When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses the latest release of version `10`, but more recent version numbers can be used instead.
```

File: docs/docs/gatsby-on-linux.md [hunk 13]
```diff
@@ -2,12 +2,126 @@
...
+
+> Note that `npm` comes packaged with `node`
+
+Finally, install `git` which will be necessary for creating your first Gatsby project based on a starter.
+
+```shell
+sudo apt install git
+```
+
+### Fedora, RedHat and other `dnf` based distros
```
Comment (marcysutton) [inline docs/docs/gatsby-on-linux.md]:
```suggestion
## Fedora, RedHat, and other `dnf` based distros
```

File: docs/docs/gatsby-on-linux.md [hunk 14]
```diff
@@ -2,12 +2,126 @@
...
+
+These distros come installed with `curl`, so you can use that to download `nvm`.
+
+```shell
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.1/install.sh | bash
+```
+
+> Note that this is the current stable release of nvm. Full installation instructions and troubleshooting can be found at the [nvm GitHub page](https://github.com/nvm-sh/nvm)
+
+When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses the latest release of version `10`, but other version numbers can be used instead.
```
Comment (marcysutton) [inline docs/docs/gatsby-on-linux.md]:
```suggestion
When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses the latest release of version `10`, but more recent version numbers can be used instead.
```

File: docs/docs/gatsby-on-linux.md [hunk 15]
```diff
@@ -2,12 +2,126 @@
...
+
+> Note that `npm` comes packaged with `node`
+
+Finally, install `git` which will be necessary for creating your first Gatsby project based on a starter.
+
+```shell
+sudo dnf install git
+```
+
+### Archlinux and other `pacman` based distros
```
Comment (marcysutton) [inline docs/docs/gatsby-on-linux.md]:
```suggestion
## Archlinux and other `pacman` based distros
```

File: docs/docs/gatsby-on-linux.md [hunk 16]
```diff
@@ -2,12 +2,126 @@
...
+
+> Note that this is the current stable release of nvm. Full installation instructions and troubleshooting can be found at the [nvm GitHub page](https://github.com/nvm-sh/nvm)
+
+Before using `nvm`, you need to install additional dependencies.
+
+```shell
+sudo pacman -S grep awk tar git
+```
+
+When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses the latest release of version `10`, but other version numbers can be used instead.
```
Comment (marcysutton) [inline docs/docs/gatsby-on-linux.md]:
```suggestion
When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses the latest release of version `10`, but more recent version numbers can be used instead.
```

File: docs/docs/gatsby-on-linux.md [hunk 17]
```diff
@@ -2,12 +2,126 @@
...
+
+```shell
+node -v
+```
+
+> Note that `npm` comes packaged with `node`
 
 ## Windows Subsystem Linux (WSL)
 
+This guide assumes that you already have WSL installed with a working Linux distro. If you don't, follow [this guide from microsoft's site](https://docs.microsoft.com/en-us/windows/wsl/install-win10) to install WSL and a Linux distro of your choice.
```
Comment (marcysutton) [inline docs/docs/gatsby-on-linux.md]:
```suggestion
This guide assumes that you already have WSL installed with a working Linux distro. If you don't, follow [this guide from Microsoft's site](https://docs.microsoft.com/en-us/windows/wsl/install-win10) to install WSL and a Linux distro of your choice.
```

**Comment by 1qk1** [comment]
I went ahead and applied @marcysutton's comments since it was just minor style improvements and typo fixes!

**Comment by laurieontech** [comment]
Awesome! Thanks for the review @marcysutton and for making those edits @1qk1!

Going to merge this in. @1qk1 are you good to port these instructions to #18262?
